common/perf-test-core/src/main/java/com/azure/perf/test/core/RepeatingInputStream.java

```diff
@@ -50,15 +64,12 @@ public synchronized int read(byte[] b, int off, int len) {
             return -1;
         }
 
-        int readCount = Math.min(len, RANDOM_BYTES_LENGTH);
-        long remaining = this.size - this.pos;
-        if (remaining < readCount) {
-            readCount = (int) remaining;
+        int read = 0;
+        for (; read < len && pos < size; read++, pos++) {
+            b[off + read] = (byte) getByte(pos);
```

Comment:
Long term we should look into re-introducing usage of `System.arraycopy` here for potentially better performance (from what I remember this can hook into intrinsics).